google/datalab/contrib/mltoolbox/_local_predict.py

```diff
@@ -158,11 +165,109 @@ def get_prediction_results(model_dir, data, headers, img_cols=None, show_image=T
   if show_image:
     display_data = _get_display_data_with_images(data, images)
 
-  predict_results = _tf_predict(model_dir, predict_data)
+  if cloud:
+    parts = model_dir_or_id.split('.')
+    if len(parts) != 2:
+      raise ValueError('Invalid model name for cloud prediction. Use "model.version".')
+
+    predict_results = ml.ModelVersions(parts[0]).predict(parts[1], predict_data)
+  else:
+    predict_results = _tf_predict(model_dir_or_id, predict_data)
+
   df_r = pd.DataFrame(predict_results)
   df_s = pd.DataFrame(display_data)
   df = pd.concat([df_r, df_s], axis=1)
   # Remove duplicate columns. All 'key' columns are duplicate here.
   df = df.loc[:, ~df.columns.duplicated()]
 
   return df
+
+
+def _batch_csv_reader(csv_file, n):
+  with file_io.FileIO(csv_file, 'r') as f:
+    args = [f] * n
+    return itertools.izip_longest(*args)
+
+
+def _get_output_schema(session, output_alias_map):
+  schema = []
+  for name in sorted(six.iterkeys(output_alias_map)):
+    tensor_name = output_alias_map[name]
+    dtype = session.graph.get_tensor_by_name(tensor_name).dtype
+    if dtype == tf.int32 or dtype == tf.int64:
+      schema.append({'name': name, 'type': 'INTEGER'})
+    elif dtype == tf.float32 or dtype == tf.float64:
+      schema.append({'name': name, 'type': 'FLOAT'})
+    else:
+      schema.append({'name': name, 'type': 'STRING'})
+
+  return schema
+
+
+def _format_results(output_format, output_schema, batched_results):
+  # Convert a dict of list to a list of dict.
+  batched_results = [dict(zip(batched_results, t)) for t in zip(*batched_results.values())]
```

Comment:
no. Nice catch! I didn't know TF returns individual values instead of list if batch size is 1. Fixed.